src/View/Widget/WidgetLocator.php

```diff
@@ -185,7 +185,7 @@ public function clear(): void
      * @return \Cake\View\Widget\WidgetInterface Widget instance.
      * @throws \ReflectionException
      */
-    protected function _resolveWidget($config): WidgetInterface
+    protected function _resolveWidget(mixed $config): WidgetInterface
```

Comment:
```suggestion
    protected function _resolveWidget(string|array $config): WidgetInterface
```

Then the exception with msg `'Widget config must be a string or array.'` can be removed.